### PR TITLE
Upgrade script

### PR DESCRIPTION
**Pull request by ivanzhelyazkov**
Add upgrade script and post-upgrade tests for CarbonController

### CODE REVIEW

**Review by lbeder — APPROVED** [review]

File: contracts/utility/MathEx.sol [hunk 1]
```diff
@@ -58,8 +58,8 @@ library MathEx {
     }
 
     /**
-    * @dev returns the smallest integer `z` such that `x * y / z <= 2 ^ 256 - 1`
-    */
+     * @dev returns the smallest integer `z` such that `x * y / z <= 2 ^ 256 - 1`
```
Comment (yudilevi) [inline contracts/utility/MathEx.sol:61]: not merged from dev?

File: deploy/scripts/0005-CarbonController-upgrade.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { DeployedContracts, InstanceName, isMainnet, setDeploymentMetadata, upgradeProxy } from '../../utils/Deploy';
```
Comment (yudilevi) [inline deploy/scripts/0005-CarbonController-upgrade.ts:1]: run linter

File: deploy/scripts/0005-CarbonController-upgrade.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+
+const func: DeployFunction = async ({ getNamedAccounts }: HardhatRuntimeEnvironment) => {
+    const { daoMultisig } = await getNamedAccounts();
+    const voucher = await DeployedContracts.Voucher.deployed();
+
+    const carbonController = await DeployedContracts.CarbonController.deployed();
+    await upgradeProxy({
+        name: InstanceName.CarbonController,
+        from: daoMultisig,
```
Comment (yudilevi) [inline deploy/scripts/0005-CarbonController-upgrade.ts:12]: I believe it should be `deployer` here

**Review by yudilevi — CHANGES_REQUESTED** [review]

File: contracts/utility/MathEx.sol (hunk 1 above)
Comment (ivanzhelyazkov) [reply contracts/utility/MathEx.sol]: fixed

File: deploy/scripts/0005-CarbonController-upgrade.ts (hunk 2 above)
Comment (ivanzhelyazkov) [reply deploy/scripts/0005-CarbonController-upgrade.ts]: fixed lint issues

File: deploy/scripts/0005-CarbonController-upgrade.ts (hunk 3 above)
Comment (lbeder) [reply deploy/scripts/0005-CarbonController-upgrade.ts]: How come the tests passed?

**Pull request merged**